File: go/wmi/WmiProperty.go
package wmi;

type WmiType string;

const (
	Int WmiType = "int";
)

type WmiPropertyFlags int;

const(
	WmiPropertyFlags None = 0,
    WmiPropertyFlags    Class = 1,
    WmiPropertyFlags    Method = 2,
    WmiPropertyFlags    Property = 4,
    WmiPropertyFlags    Parameter = 8,
	WmiPropertyFlags    Association = 16,
	WmiPropertyFlags    Indication = 32,
	WmiPropertyFlags   Reference = 64,
	WmiPropertyFlags    Any = 127,
	WmiPropertyFlags   EnableOverride = 128,
	WmiPropertyFlags   DisableOverride = 256,
	WmiPropertyFlags    Restricted = 512,
	WmiPropertyFlags    ToSubclass = 1024,
	WmiPropertyFlags    Translatable = 2048,
	WmiPropertyFlags    Key = 4096,
	WmiPropertyFlags    In = 8192,
	WmiPropertyFlags   Out = 16384,
	WmiPropertyFlags    Required = 32768,
	WmiPropertyFlags    Static = 65536,
	WmiPropertyFlags   Abstract = 131072,
	WmiPropertyFlags    Terminal = 262144,
	WmiPropertyFlags    Expensive = 524288,
	WmiPropertyFlags   Stream = 1048576,
	WmiPropertyFlags    ReadOnly = 2097152,
	WmiPropertyFlags   NotModified = 33554432,
	WmiPropertyFlags    NullValue = 536870912,
	WmiPropertyFlags    Borrow = 1073741824,
	//WmiPropertyFlags Adopt = 2147483648,
)

type WmiProperty interface {
	Name() string;
	Value() string;
	Type() WmiType;
	Flags() WmiPropertyFlags;
}

File: dotnet/wmilib/interface/WmiProperty.cs
// Copyright 2019 (c) Microsoft Corporation.
// Licensed under the MIT license.

namespace Microsoft.WmiCodeGen.DotNet.Interface
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    public interface IWmiProperty
    {
        #region Properties

        /// <summary>
        /// Gets the name.
        /// </summary>
        /// <value>
        /// The name.
        /// </value>
        string Name
        {
            get;
        }

        /// <summary>
        /// Gets or sets the value.
        /// </summary>
        /// <value>
        /// The value.
        /// </value>
        object Value
        {
            get;
            set;
        }

        /// <summary>
        /// Gets the type.
        /// </summary>
        /// <value>
        /// The type.
        /// </value>
        Type Type
        {
            get;
        }

        WmiPropertyFlags Flags
        {
            get;
        }

        #endregion Properties
    }

    [Flags]
    public enum WmiPropertyFlags
    {
        None = 0,
        Class = 1,
        Method = 2,
        Property = 4,
        Parameter = 8,
        Association = 16,
        Indication = 32,
        Reference = 64,
        Any = 127,
        EnableOverride = 128,
        DisableOverride = 256,
        Restricted = 512,
        ToSubclass = 1024,
        Translatable = 2048,
        Key = 4096,
        In = 8192,
        Out = 16384,
        Required = 32768,
        Static = 65536,
        Abstract = 131072,
        Terminal = 262144,
        Expensive = 524288,
        Stream = 1048576,
        ReadOnly = 2097152,
        NotModified = 33554432,
        NullValue = 536870912,
        Borrow = 1073741824,
        //Adopt = 2147483648,
    }
}
